text/1099-renderComponent.md

```diff
@@ -0,0 +1,389 @@
+---
+stage: accepted
+start-date: 2025-05-01T00:00:00.000Z
+release-date: # In format YYYY-MM-DDT00:00:00.000Z
+release-versions:
+teams: # delete teams that aren't relevant
+  - framework
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/1099 
+project-link:
+suite: 
+---
+
+<!--- 
+Directions for above: 
+
+stage: Leave as is
+start-date: Fill in with today's date, 2032-12-01T00:00:00.000Z
+release-date: Leave as is
+release-versions: Leave as is
+teams: Include only the [team(s)](README.md#relevant-teams) for which this RFC applies
+prs:
+  accepted: Fill this in with the URL for the Proposal RFC PR
+project-link: Leave as is
+suite: Leave as is
+-->
+
+<!-- Replace "RFC title" with the title of your RFC -->
+
+# `renderComponent` 
+
+## Summary
+
+Add a new API to for rendering components into any DOM element.
+
+## Motivation
+
+Building an app from scratch requires a fair amount of boilerplate today, and `renderComponent` abstracts a minimal API that would allow easy integration of components in other enviroments, such as [Astro](https://astro.build/), or [Vitepress](https://vitepress.dev/), or [Storybook](https://storybook.js.org/), or runtime demos such as [JSBin](https://jsbin.com/).
+
+This also pairs well with [RFC#931: JS Representation of Template Tag](https://github.com/emberjs/rfcs/pull/931), in that both features together give folks enough to use Ember without compilation (compilation still recommended for optimizations, however).
+
+
+## Detailed design
+
+Users would import `renderComponent` from `@ember/renderer` (a pre-existing module)
+
+The interface:
+```ts
+/**
+ * Renders a component into an element, given a valid component definition.
+ */
+export function renderComponent(
+    /**
+     * The component definition to render.
+     *
+     * Any component that has had its manager registered is valid.
+     * For the component-types that ship with ember, manager registration 
+     * does not need to be worried about. 
+     */
+    component: object,
+    options: {
+        /**
+         * The element to render the component in to.
+         */
+        into: Element;
+
+        /**
+         * Optional owner. Defaults to `{}`, can be any object, but will need to implement the [Owner](https://api.emberjs.com/ember/release/classes/Owner) API for components within this render tree to access services.
+         */
+        owner?: object;
+
+        /**
+         * Configure the `document` and `isInteractive`
+         */
+        env?: { 
+            /**
+             * Defaults to globalThis.document.
+             */
+            document?: SimpleDocument | Document; 
```

Comment:
TODO: investigate what all is used on the document -- can the interface be more minimal?